code/hud/hudsquadmsg.cpp

```diff
@@ -485,6 +487,86 @@ void hud_squadmsg_page_up()
 	}
 }
 
+//Fuctions that allow selection of specific comms menu items with simple up/down/select buttons
+void hud_squadmsg_selection_move_down() {
+
+	//Check if comms menu is up
+	if (Player->flags & PLAYER_FLAGS_MSG_MODE)
+	{
+		//move down
+		++Selected_menu_item;
+
+		//play scrolling sound and reset the comms window timeout timer, so the window doesn't dissapear while we select our item
+		gamesnd_play_iface(InterfaceSounds::SCROLL);
+		Msg_mode_timestamp = timestamp(DEFAULT_MSG_TIMEOUT);
+
+		//Move to next page if we went outside of current one
+		if (Selected_menu_item == MAX_MENU_DISPLAY 
+			&& (First_menu_item + MAX_MENU_DISPLAY < Num_menu_items))
+		{
+			hud_squadmsg_page_down();
+			Selected_menu_item = 0;
+		}
+
+		//Select the first menu item if we went outside items range, so we can loop around
+		if (First_menu_item + Selected_menu_item >= Num_menu_items) 
+		{
+			First_menu_item = 0;
+			Selected_menu_item = First_menu_item;
+		}
+	}
+}
+
+void hud_squadmsg_selection_move_up() {
+
+	//Check if comms menu is up
+	if (Player->flags & PLAYER_FLAGS_MSG_MODE)
+	{
+		//move down
+		--Selected_menu_item;
+
+		//play scrolling sound and reset the comms window timeout timer, so the window doesn't dissapear while we select our item
+		gamesnd_play_iface(InterfaceSounds::SCROLL);
+		Msg_mode_timestamp = timestamp(DEFAULT_MSG_TIMEOUT);
+
+		//Move to previous page if it exists
+		if (Selected_menu_item < 0 && First_menu_item > 0)
+		{
+			hud_squadmsg_page_up();
+			Selected_menu_item = MAX_MENU_DISPLAY - 1; //if we're moving to previous page in the first place, we assume it was already populated to the max
+		}
+
+		//Select the last menu item if we went outside items range, so we can loop around
+		else if (Selected_menu_item < 0) 
+		{
+			First_menu_item = ((Num_menu_items - 1) / 10) * 10; 
```

Comment:
Caution: integer arithmetic.  Possible abuse of hardcoded magic numbers.  What are the assumptions made here?  Please document either here or at the declaration of First_menu_item.